From be12957f23c0501cc4aae61a27048baa60d5d70a Mon Sep 17 00:00:00 2001
From: Mousumi Jana <mousumix.jana@intel.com>
Date: Thu, 26 Nov 2015 02:36:16 +0530
Subject: [PATCH 1819/2508] ASoC: Intel: Skylake:Add support to get platform
 data

commit 9e67019e63f3443dca09ef98582955734e1485e6 from
https://github.com/01org/linux-apollolake-i

adding support to get the mailbox offsets and sram
address as these are required for debugfs.

Change-Id: Ic5c6004ac12de2a342a9a21495e4c9c56efdb5e1
Signed-off-by: Mousumi Jana <mousumix.jana@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/7603
Reviewed-by: Babu, Ramesh <ramesh.babu@intel.com>
Tested-by: Babu, Ramesh <ramesh.babu@intel.com>
---
 sound/soc/intel/common/sst-dsp-priv.h |    4 ++++
 sound/soc/intel/skylake/bxt-sst.c     |    4 ++++
 sound/soc/intel/skylake/skl-debug.c   |   10 +++++++++-
 sound/soc/intel/skylake/skl.c         |   20 +++++++++++++++++++-
 sound/soc/intel/skylake/skl.h         |   18 ++++++++++++++++--
 5 files changed, 52 insertions(+), 4 deletions(-)

diff --git a/sound/soc/intel/common/sst-dsp-priv.h b/sound/soc/intel/common/sst-dsp-priv.h
index 538e917..c790540 100644
--- a/sound/soc/intel/common/sst-dsp-priv.h
+++ b/sound/soc/intel/common/sst-dsp-priv.h
@@ -77,6 +77,10 @@ struct sst_addr {
 	u32 dram_offset;
 	u32 dsp_iram_offset;
 	u32 dsp_dram_offset;
+	u32 sram0_base;
+	u32 sram1_base;
+	u32 w0_stat_sz;
+	u32 w0_up_sz;
 	void __iomem *lpe;
 	void __iomem *shim;
 	void __iomem *pci_cfg;
diff --git a/sound/soc/intel/skylake/bxt-sst.c b/sound/soc/intel/skylake/bxt-sst.c
index ea4ae0f..80453ae 100644
--- a/sound/soc/intel/skylake/bxt-sst.c
+++ b/sound/soc/intel/skylake/bxt-sst.c
@@ -110,6 +110,10 @@ int bxt_sst_dsp_init(struct device *dev, void __iomem *mmio_base, int irq,
 	sst->fw_ops = bxt_fw_ops;
 	sst->addr.lpe = mmio_base;
 	sst->addr.shim = mmio_base;
+	sst->addr.sram0_base = BXT_ADSP_SRAM0_BASE;
+	sst->addr.sram1_base = BXT_ADSP_SRAM1_BASE;
+	sst->addr.w0_stat_sz = BXT_ADSP_W0_STAT_SZ;
+	sst->addr.w0_up_sz = BXT_ADSP_W0_UP_SZ;
 
 	sst_dsp_mailbox_init(sst, (BXT_ADSP_SRAM0_BASE + BXT_ADSP_W0_STAT_SZ),
 			BXT_ADSP_W0_UP_SZ, BXT_ADSP_SRAM1_BASE, BXT_ADSP_W1_SZ);
diff --git a/sound/soc/intel/skylake/skl-debug.c b/sound/soc/intel/skylake/skl-debug.c
index f322559..43fd4da 100644
--- a/sound/soc/intel/skylake/skl-debug.c
+++ b/sound/soc/intel/skylake/skl-debug.c
@@ -60,6 +60,9 @@ struct skl_debug {
 	struct dentry *modules;
 	u32 ipc_data[MAX_SZ];
 	struct fw_ipc_data fw_ipc_data;
+	void __iomem *in_base;
+	size_t inbx_sz;
+	u32 w0_stat_sz;
 };
 
 struct nhlt_specific_cfg
@@ -659,7 +662,8 @@ static int skl_init_adsp(struct skl_debug *d)
 	return 0;
 }
 
-struct skl_debug *skl_debugfs_init(struct skl *skl)
+struct skl_debug *skl_debugfs_init(struct skl *skl,
+					struct platform_info *dbg_info)
 {
 	struct skl_debug *d;
 
@@ -678,6 +682,9 @@ struct skl_debug *skl_debugfs_init(struct skl *skl)
 
 	d->skl = skl;
 	d->dev = &skl->pci->dev;
+	d->in_base = dbg_info->in_base;
+	d->inbx_sz = dbg_info->in_size;
+	d->w0_stat_sz = dbg_info->w0_stat_sz;
 
 	/* now create the NHLT dir */
 	d->nhlt =  debugfs_create_dir("nhlt", d->fs);
@@ -696,6 +703,7 @@ struct skl_debug *skl_debugfs_init(struct skl *skl)
 	skl_init_nhlt(d);
 	skl_init_adsp(d);
 	skl_init_mod_set_get(d);
+	kfree(dbg_info);
 
 	return d;
 
diff --git a/sound/soc/intel/skylake/skl.c b/sound/soc/intel/skylake/skl.c
index 57ccc84..8a7acae 100644
--- a/sound/soc/intel/skylake/skl.c
+++ b/sound/soc/intel/skylake/skl.c
@@ -30,6 +30,8 @@
 #include <sound/hdaudio.h>
 #include <sound/hda_i915.h>
 #include "skl.h"
+#include "../common/sst-dsp-priv.h"
+#include "skl-sst-ipc.h"
 
 static char *machine = NULL;
 
@@ -567,6 +569,7 @@ static int skl_probe(struct pci_dev *pci,
 	struct skl *skl;
 	struct hdac_ext_bus *ebus = NULL;
 	struct hdac_bus *bus = NULL;
+	struct platform_info *dbg_info;
 	int err;
 
 	/* we use ext core ops, so provide NULL for ops here */
@@ -620,8 +623,23 @@ static int skl_probe(struct pci_dev *pci,
 	if (err < 0)
 		goto out_unregister;
 
+	dbg_info = kzalloc(sizeof(struct platform_info), GFP_KERNEL);
+	if (!dbg_info)
+		return -ENOMEM;
+
+	dbg_info->sram0_base = skl->skl_sst->dsp->addr.sram0_base;
+	dbg_info->sram1_base = skl->skl_sst->dsp->addr.sram1_base;
+	dbg_info->lpe = skl->skl_sst->dsp->addr.lpe;
+	dbg_info->w0_stat_sz = skl->skl_sst->dsp->addr.w0_stat_sz;
+	dbg_info->w0_up_sz = skl->skl_sst->dsp->addr.w0_up_sz;
+
+	dbg_info->in_base = skl->skl_sst->dsp->mailbox.in_base;
+	dbg_info->in_size = skl->skl_sst->dsp->mailbox.in_size;
+	dbg_info->out_base = skl->skl_sst->dsp->mailbox.out_base;
+	dbg_info->out_size = skl->skl_sst->dsp->mailbox.out_size;
+
 	/* init debugfs */
-	skl->debugfs = skl_debugfs_init(skl);
+	skl->debugfs = skl_debugfs_init(skl, dbg_info);
 
 	/*configure PM */
 	pm_runtime_set_autosuspend_delay(bus->dev, SKL_SUSPEND_DELAY);
diff --git a/sound/soc/intel/skylake/skl.h b/sound/soc/intel/skylake/skl.h
index 9f82eea..54c3914 100644
--- a/sound/soc/intel/skylake/skl.h
+++ b/sound/soc/intel/skylake/skl.h
@@ -78,6 +78,18 @@ struct skl {
 	bool mod_set_get_status;
 };
 
+struct platform_info {
+	u32 sram0_base;
+	u32 sram1_base;
+	u32 w0_stat_sz;
+	u32 w0_up_sz;
+	size_t in_size;
+	size_t out_size;
+	void __iomem *lpe;
+	void __iomem *in_base;
+	void __iomem *out_base;
+};
+
 #define skl_to_ebus(s)	(&(s)->ebus)
 #define ebus_to_skl(sbus) \
 	container_of(sbus, struct skl, sbus)
@@ -115,7 +127,8 @@ struct skl_module_cfg;
 
 #ifdef CONFIG_DEBUG_FS
 
-struct skl_debug *skl_debugfs_init(struct skl *skl);
+struct skl_debug *skl_debugfs_init(struct skl *skl,
+					struct platform_info *dbg_info);
 void skl_debugfs_exit(struct skl_debug *d);
 struct nhlt_specific_cfg
 *skl_nhlt_get_debugfs_blob(struct skl_debug *d, u8 link_type, u32 instance,
@@ -128,7 +141,8 @@ void skl_debug_init_module(struct skl_debug *d,
 struct skl_debug {
 }
 
-struct skl_debug *skl_debugfs_init(struct skl *skl)
+struct skl_debug *skl_debugfs_init(struct skl *skl,
+					struct platform_info *dbg_info)
 {
 	return NULL;
 }
-- 
1.7.5.4

